### PR TITLE
[Glimmer2] Cleanup input tests

### PR DESCRIPTION
**Pull request by chadhietala**
This will supersede #13222

### CODE REVIEW

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} is set on noop rerender`);
+
+    this.setComponentValue(values[1]);
+    this.assertAttributeHasValue(attributeName, values[1], `${attributeName} is set on rerender`);
+
+    this.setComponentValue(values[0]);
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} can be set back to the initial value`);
+  }
+
+  ['@test input disabled attribute']() {
```
Comment (stefanpenner) [inline packages/ember-glimmer/tests/integration/input-test.js]: why do we wrap these in brackets? leaving them off is valid es6 syntax i think.

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
+import { RenderingTest, moduleFor } from '../utils/test-case';
+import { set } from 'ember-metal/property_set';
+
+moduleFor('Input element tests', class extends RenderingTest {
+  runAttributeTest(attributeName, values) {
+    let template = `<input ${attributeName}={{value}}>`;
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: Did you intentionally not quote or could it also have been `="{{value}}"` instead?

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+  ['@test input disabled attribute']() {
+    this.runPropertyTest('disabled', [false, true]);
+  }
+
+  ['@test input value attribute']() {
+    this.runPropertyTest('value', ['foo', 'bar']);
+  }
+
+  ['@htmlbars input placeholder attribute']() {
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: What is preventing this from working in glimmer?

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+  ['@htmlbars input maxlength attribute']() {
+    this.runAttributeTest('maxlength', [2, 3]);
+  }
+
+  ['@test input size attribute']() {
+    this.runAttributeTest('size', [2, 3]);
+  }
+
+  ['@htmlbars input tabindex attribute']() {
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: Same as https://github.com/emberjs/ember.js/pull/13392/files#r60971131.

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  ['@htmlbars input placeholder attribute']() {
+    this.runAttributeTest('placeholder', ['foo', 'bar']);
+  }
+
+  ['@test input name attribute']() {
+    this.runAttributeTest('name', ['nam', 'name']);
+  }
+
+  ['@htmlbars input maxlength attribute']() {
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: Same as https://github.com/emberjs/ember.js/pull/13392/files#r60971131.

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    this.setDOMValue('hello');
+    this.setSelectionRange(1, 3);
+
+    this.setComponentValue('hello');
+
+    this.assertSelectionRange(1, 3);
+
+    // Note: We should eventually get around to testing reseting, however
+    // browsers handle `selectionStart` and `selectionEnd` differently
+    // when are synthetically testing movement of the cursor.
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: :+1:

File: packages/ember-glimmer/tests/integration/input-test.js (hunk 3 above)
Comment (chadhietala) [inline packages/ember-glimmer/tests/integration/input-test.js]: I'm actually not sure, in IE9 this fails and I think it has to do with attributes vs. properties.

File: packages/ember-glimmer/tests/integration/input-test.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+    this.assertValue('hello', 'Value is initialised');
+
+    this.setDOMValue('hola');
+
+    this.assertValue('hola', 'DOM is used');
+
+    this.setComponentValue('hello');
+
+    this.assertValue('hola', 'DOM is used');
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: This is not correct. It should be `hello`. While it is true that the DOM is the SSOT, this doesn't mean that mutation via `setComponentValue` cannot mutate the SSOT.

File: packages/ember-glimmer/tests/integration/input-test.js (hunk 2 above)
Comment (chadhietala) [inline packages/ember-glimmer/tests/integration/input-test.js]: We could do both 😄 . Just trying to get this carried over the line.

File: packages/ember-glimmer/tests/integration/input-test.js (hunk 7 above)
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: This should pass if you make it @htmlbars for now.

File: packages/ember-glimmer/tests/integration/input-test.js (hunk 2 above)
Comment (mmun) [inline packages/ember-glimmer/tests/integration/input-test.js]: I think it's fine as is.

File: packages/ember-glimmer/tests/integration/input-test.js (hunk 7 above)
Comment (chadhietala) [inline packages/ember-glimmer/tests/integration/input-test.js]: `@skip` for now.

**Pull request merged**